### ISSUE TITLE
Are you still using/developing this?

### ISSUE THREAD

**Issue by Frans-Willem** [patman15/hdpv_ble#1]
### Checklist

- [X] I need support with using the integration.
- [X] I'm not avoiding to fill out the [bug report](https://github.com/patman15/hdpv_ble/issues/new?assignees=&labels=Bug&projects=&template=bug.yml) form.

### Describe the issue

Hi,

Are you still developing this, and if so, are you open to working together to reverse engineer more of the protocol ?

Thanks,
Frans-Willem

**Comment by patman15** [comment]
Yes, the project is still active, it's working for me quite well and I wanted to add tests for quality. I failed to find a way to add HA as a "remote" in order to have a user-friendly way to get the encryption key.
Saw your PR with the script, thanks for that, but I would expect users of this integration not to have a gateway ...
Anyway, I'm happy to follow up with more features, or PRs, are you looking for something specific?

**Comment by Frans-Willem** [comment]
Did you ever get the ESP32 PowerView emulator working better? I think that'd be a pretty user-friendly way to intercept the homekey, but I've been able to get it to receive data from the android app, but not send a response.

I've reverse engineered the PowerView Android APK and have been able to document a lot more about the protocol and the APIs available on the gateway and the cloud, but haven't found a better way to get the homekey. I was hoping there'd be something in the cloud API to get the homekey.

**Comment by patman15** [comment]
Regarding the key my best idea was to emulate a remote and then add it to the app.. Unfortunately, the remote seems to use the L2CAP protocol instead of GATT so the HA libraries do not support it. Due to lack of time I did not proceed in emulating a remote. I can emulate a shade so you could use an ESP32 to emulate the shade, add it in the app, which will give you the key. Nevertheless, you need an ESP32 for that. B-)

I am aware that the shades support a lot more functions, but I didn't find a good reason to add them. First, things, like schedules, can be easily done in Home Assistant itself with the current functionality and second because HA has no natural interface for them.

If you have a good use case, let me know, can be done of course.

Best regards,
Patrick

**Comment by Frans-Willem** [comment]
I'd preferably like to emulate a shade to catch the SetShadeKey call, however your current example has trouble responding to calls.
The current code seems to attempt to respond to the ResetSceneAutomations call, but it looks like the android app never receives that response.
Were you ever able to figure out why that was not working?

**Comment by patman15** [comment]
I think I fixed this issue in my local repository but did not push it to GitHub because the code was a mess. B-) I didn't really polish it because I was hoping to find an even better solution and so far nobody asked.

Shouldn't be an issue to do an ESP32 sketch that does what you would like to see. Just need to ask a bit of patience from you, because I only have access to my development environment at the end of next week again.

**Comment by Frans-Willem** [comment]
Would be very interested in the updated simulator :)

**Comment by patman15** [comment]
Sorry for the delay, just checked, unfortunately, I had the latest state published. That means, I have no newer version, but I can fix it to get it working. So let me get back to you with a better working version or if you have PRs to integrate, happy to take them in as well!

**Comment by patman15** [comment]
@Frans-Willem dedicating a bit of time to fixing the emulator, but hit some difficulties. Just saying, I did not forget about you. :wink:

**Comment by gschot** [comment]
Came across this integration, and my apologies for barging into a running thread. I just have some shades (TDBU) installed and I'm running them via the Powerview Gen3 hub. This integration would be quite interesting though as I already have ESP32s throughout the house for BT coverage. The shades show up nicely in the integration, but only BU. The shades I have report being type 8. Any plans to add more types to the supported list?

**Comment by patman15** [comment]
> The shades I have report being type 8. Any plans to add more types to the supported list?
> 

No plans, but if you are willing to dedicate some time for testing (I do not own that type of shades) and in general are a bit patient with progress, we can make plans. :-)
If you want to work together to get it integrated, please open a new issue and let's align there.
